Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/extrude.h

```diff
@@ -0,0 +1,304 @@
+  copy_face_graph(input, output, std::back_inserter(bottom_v2v),
+                        std::back_inserter(bottom_h2h), Emptyset_iterator(),
+                  input_vpm, output_vpm);
+  
+  // create the offset for the other side
+  for(std::size_t i = 0; i< bottom_v2v.size(); ++i)
+  {
+    bot(bottom_v2v[i].first, bottom_v2v[i].second);
+  }
+  CGAL::Polygon_mesh_processing::reverse_face_orientations(output);
+  
+  // collect border halfedges for the creation of the triangle strip
+  std::vector<std::pair<input_vertex_descriptor, output_vertex_descriptor> > top_v2v;
+  std::vector<std::pair<input_halfedge_descriptor, output_halfedge_descriptor> > top_h2h;
+  copy_face_graph(input, output, std::inserter(top_v2v, top_v2v.end()),
+                        std::inserter(top_h2h, top_h2h.end()), Emptyset_iterator(),
+                  input_vpm, output_vpm);
+  for(std::size_t i = 0; i< top_v2v.size(); ++i)
+  {
+    top(top_v2v[i].first, top_v2v[i].second);
+  }
+  std::vector<output_halfedge_descriptor> border_hedges;
+  std::vector<output_halfedge_descriptor> offset_border_hedges;
+  for(std::size_t i = 0; i< top_h2h.size(); ++i)
+  {
+    input_halfedge_descriptor h = top_h2h[i].first;
+    if( CGAL::is_border(h, input) )
+    {
+      border_hedges.push_back(top_h2h[i].second);
+      offset_border_hedges.push_back(bottom_h2h[i].second);
+      CGAL_assertion(is_border(border_hedges.back(), output));
+      CGAL_assertion(is_border(offset_border_hedges.back(), output));
+    }
+  }
+  // now create a quad strip
+  for(std::size_t i=0; i< border_hedges.size(); ++i)
+  {
+    output_halfedge_descriptor h1 = border_hedges[i], h2=offset_border_hedges[i],
+        nh1 = next(h1, output), ph2 = prev(h2, output);
+    output_halfedge_descriptor newh = halfedge(add_edge(output), output),
+        newh_opp = opposite(newh, output);
+    // set target vertices of the new halfedges
+    set_target(newh, target(h1, output), output);
+    set_target(newh_opp, target(ph2, output), output);
+    // update next/prev pointers
+    set_next(h1, newh_opp, output);
+    set_next(newh_opp, h2, output);
+    set_next(ph2, newh, output);
+    set_next(newh, nh1, output);
+  }
+  
+  // create new faces
+  for(std::size_t i=0; i< border_hedges.size(); ++i)
+  {
+    output_halfedge_descriptor h = border_hedges[i];
+    
+    output_face_descriptor nf = add_face(output);
+    
+    CGAL::cpp11::array<output_halfedge_descriptor, 4> hedges;
+    for (int k=0; k<4; ++k)
+    {
+      hedges[k]=h;
+      h = next(h, output);
+    }
+    
+    set_face(hedges[0], nf, output);
+    set_face(hedges[1], nf, output);
+    set_face(hedges[2], nf, output);
+    set_face(hedges[3], nf, output);
+    set_halfedge(nf, hedges[0], output);
+    Euler::split_face(hedges[0], hedges[2], output);
+  }
+}
+
+
+/**
+ * \ingroup PMP_meshing_grp
+ * Extrudes `input` into `output` following the direction given by `dir` and
+ * at a distance `d`.
```

Comment:
I'd add something like that:

fills `output` with a close mesh bounding the volume swept by `input` when translating its vertices by `dir*d`. The mesh is oriented so that the faces corresponding to `input` in `output` have the same orientation.